src/test/java/use_cases/PlayerGameInteractorTest.java

```diff
@@ -0,0 +1,221 @@
+        players = new ArrayList<>();
+        players.add(p1);
+        players.add(p2);
+        players.add(p3);
+        game = new Game(deck, players);
+        // Make the first Card of the game something p1 can play on top of.
+        firstCard = new Card(h1.getCards().get(0).getSuit(), "King");
+        game.putCardDown(firstCard);
+        gameState = new GameState(game);
+        game.addObserver(gameState);
+        String p1CardVal = h1.getCards().get(0).getValue();
+        String p1CardSuit = h1.getCards().get(0).getSuit();
+        p1sCard = h1.getCards().get(0);
+        interactor = new PlayerGameInteractor(game, gameState);
+        playCardReq = new PlayerGameRequestModel("sol", p1CardVal, p1CardSuit, true, false, false);
+        pickUpReq = new PlayerGameRequestModel("sol", null, null, false, true, false);
+        skipReq = new PlayerGameRequestModel("sol", null, null, false, false, true);
+
+    }
+
+    /**
+     * Set all made objects in setUp() to null to free memory.
+     */
+    @AfterEach
+    void tearDown() {
+        gameState = null;
+        p1 = null;
+        p2 = null;
+        game = null;
+        players = null;
+        interactor = null;
+        playCardReq = null;
+        skipReq = null;
+        pickUpReq = null;
+        p1sCard = null;
+        firstCard = null;
+    }
+
+    /**
+     * Test createResponse() when given the input to play a Card and the Card is valid.
+     */
+    @Test
+    public void testCreateResponsePlayValidCard() {
+        PlayerGameResponseModel response = interactor.createResponse(playCardReq);
+        // Assert turn has been changed to p3 (p2 is Computer, so logic should be done)
+        assertEquals(p3.getName(), response.getCurrentPlayerName());
+        // Assert that p1 has lost a Card and that Card is in the deck.
+        assertEquals(4, p1.getNumCards());
+        assertTrue(game.getGameDeck().getCards().contains(p1sCard));
+        // Assert the Game has no winner yet.
+        assertFalse(response.getHasWinner());
+    }
+
+    /**
+     * Test createResponse() when given the input to play a Card and the Card is invalid.
+     */
+    @Test
+    public void testCreateResponsePlayInvalidCard() {
+        // Give a new bogus Hand to p1 so to validate p1 has no valid Cards.
+        Hand newHand = new Hand(new ArrayList<>());
+        Card bogus1 = new Card("Test", "17");
+        Card bogus2 = new Card("Test", "18");
+        newHand.addCard(bogus1);
+        newHand.addCard(bogus2);
+        p1.setHand(newHand);
+        String chosenSuit = bogus1.getSuit();
+        String chosenValue = bogus1.getValue();
+        // Have p1 try to play bogus1.
+        PlayerGameRequestModel pgrm = new PlayerGameRequestModel("sol", chosenValue, chosenSuit, true, false, false);
+        interactor.createResponse(pgrm);
+        // Assert the turn has not changed, no winner, current Card is the same, none of the player's cards have changed.
```

Comment:
Maybe I missed this, but what happens in the event that the user tries to play an invalid card? Does a message show up?

Reply:
If the user tries to play an invalid card, nothing happens (well a lot of logic does) but everything remains the same. No message comes up.

Reply:
A Player will have to pick up their own cards (press the button), the game doesn't automatically pick up cards for them.

Reply:
Got it, maybe we can implement something like letting the user know if it's an invalid card to play later down the line. That way there's less barriers for new players to understand the game (accessibility stuff).